homeassistant/components/sonos/diagnostics.py

```diff
@@ -47,11 +47,11 @@ async def async_get_config_entry_diagnostics(
     hass: HomeAssistant, config_entry: ConfigEntry
 ) -> dict[str, Any]:
     """Return diagnostics for a config entry."""
-    payload = {"current_timestamp": time.monotonic()}
+    payload: dict[str, Any] = {"current_timestamp": time.monotonic()}
 
     for section in ("discovered", "discovery_known"):
         payload[section] = {}
-        data = getattr(hass.data[DATA_SONOS], section)
+        data: dict[str, Any] = getattr(hass.data[DATA_SONOS], section)
```

Comment:
```suggestion
        data: set[Any] | dict[str, Any] = getattr(hass.data[DATA_SONOS], section)
```
At least based on the `isinstance(data, set)` call directly below.